scalingo/resource_scalingo_domain.go

```diff
@@ -52,21 +59,51 @@ func resourceDomainCreate(ctx context.Context, d *schema.ResourceData, meta inte
 	return nil
 }
 
-func resourceDomainRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func resourceDomainUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, _ := meta.(*scalingo.Client)
 
 	appID, _ := d.Get("app").(string)
+	canonical, _ := d.Get("canonical").(bool)
+
+	if d.HasChange("canonical") {
+		var domain scalingo.Domain
+		var err error
+		if canonical {
+			domain, err = client.DomainSetCanonical(ctx, appID, d.Id())
+		} else {
+			// This may cause an issue when the user is changing which domain is canonical
+			// This may add the new canonical flag first and remove the old one after
+			// In this case the newly set canonical flag will be removed instead of the old one
```

Comment:
I don't really understand this last line.

If another domain is set canonical first, this one will still be `Unset` no?

Reply:
As the canonical unset is global, if we set a new canonical domain first and then remove the old canonical domain we will encounter this situation : 
![Screenshot from 2022-09-16 16-56-30](https://user-images.githubusercontent.com/112880007/190669107-7e0eabb5-42d6-473a-b04f-2ed4d89c7f3d.png)
![Screenshot from 2022-09-16 16-56-51](https://user-images.githubusercontent.com/112880007/190669116-aca21faa-4443-48d4-9364-551152c40ce8.png)